src/Spinners/LoadSpinnerText.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {
+  FunctionComponent, ReactElement, useContext,
+} from 'react';
+import styled, { ThemeContext } from 'styled-components';
+import { BaseTheme } from '../Theme';
+
+export interface LoadSpinnerTextProps {
+  /** Specifies the text under the spinner */
+  children: string;
+  /** Specifies the color of the text under the spinner */
+  color?: string;
+}
+
+const StyledLoadSpinnerText = styled.div`
+  background: transparent;
+  color: ${({ color }): string => (color)};
+  font: ${({ theme }): string => (theme.font.heading)};
+  margin-top: ${({ theme }): string => (theme.ws.large)};
+  margin-bottom: ${({ theme }): string => (theme.ws.medium)};
+  display: block;
+`;
+
+/**
+ * Data load spinner component with styled text under the spinner
+ */
+const LoadSpinnerText: FunctionComponent<LoadSpinnerTextProps> = (props):
+ReactElement => {
+  const { children, color } = props;
+  const theme: BaseTheme = useContext(ThemeContext);
+  return (
+    <StyledLoadSpinnerText theme={theme} color={color}>
+      { children }
+    </StyledLoadSpinnerText>
+  );
+};
```

Comment:
When we're directly mapping `props` and `children` to a styled component like this, it's much cleaner to just use:

```tsx
const LoadSpinnerText:
ForwardRefExoticComponent<LoadSpinnerTextProps> = withTheme(StyledLoadSpinnerText);

declare type LoadSpinnerText = ReactElement<LoadSpinnerProps>;

export default LoadSpinnerText;
```

(I've found that all the extra `type` annotations are needed to get the styleguide to properly recognize our components and their props)

Reply:
I did not know there is such way to define a component.

Reply:
Do you have documentation to help me understand how does your code work

Reply:
OK, that's totally fine to leave this as is.

Just to give a little more explanation (which I should have done from the beginning, sorry): The [`withTheme` function](https://styled-components.com/docs/api#withtheme) is a higher-order component (a function that returns a new component) that's part of the the `styled-components` library that automatically injects the theme into the `theme` prop. It ends up generating the same output, but has the advantage of not requiring you to manually pass through props. In this case it's not a big deal, but there were some other components where we were passing though a number of different props and it was getting to be more of a mess.

Reply:
Oh, and as a bit of clarification on the type annotations:

- The `ForwardRefExoticComponent<LoadSpinnerTextProps>` is what's being returned from the `withTheme(StyledLoadSpinnerText)` call, and that needs to be in there to allow our styleguide tool to properly parse and associate the props with the component in the final output. 

- The `ReactElement<LoadSpinnerProps>` is a less-specific type, and declaring it that way lets us use this default export as a type itself. This is particularly useful with something like our `Table` components, where we're only allowing certain components to be used as `children`, though that's not really the case here.